idaes/gas_solid_contactors/properties/oxygen_iron_OC_oxidation/hetero_reactions.py

```diff
@@ -0,0 +1,517 @@
+                           value,
+                           Var)
+from pyomo.util.calc_var_value import calculate_variable_from_constraint
+from pyomo.opt import SolverFactory
+from pyomo.common.config import ConfigBlock, ConfigValue, In
+
+
+# Import IDAES cores
+from idaes.core import (declare_process_block_class,
+                        MaterialFlowBasis,
+                        ReactionParameterBlock,
+                        ReactionBlockDataBase,
+                        ReactionBlockBase,
+                        Component,
+                        VaporPhase,
+                        SolidPhase)
+from idaes.core.util.misc import add_object_reference
+from idaes.core.util.initialization import (fix_state_vars,
+                                            revert_state_vars,
+                                            solve_indexed_blocks)
+from idaes.core.util.model_statistics import (
+    number_unfixed_variables_in_activated_equalities)
+from idaes.core.util.config import (is_state_block,
+                                    is_physical_parameter_block,
+                                    is_reaction_parameter_block)
+import idaes.logger as idaeslog
+
+# Some more information about this module
+__author__ = "Chinedu Okoli"
+
+
+# Set up logger
+_log = idaeslog.getLogger(__name__)
+
+
+@declare_process_block_class("HeteroReactionParameterBlock")
+class ReactionParameterData(ReactionParameterBlock):
+    """
+    Property Parameter Block Class
+
+    Contains parameters and indexing sets associated with properties for
+    superheated steam.
+
+    """
+    # Create Class ConfigBlock
+    CONFIG = ConfigBlock()
+    CONFIG.declare("gas_property_package", ConfigValue(
+            description="Reference to associated PropertyPackageParameter "
+                        "object for the gas phase.",
+            domain=is_physical_parameter_block))
+    CONFIG.declare("solid_property_package", ConfigValue(
+            description="Reference to associated PropertyPackageParameter "
+                        "object for the solid phase.",
+            domain=is_physical_parameter_block))
+    CONFIG.declare("default_arguments", ConfigBlock(
+            description="Default arguments to use with Property Package",
+            implicit=True))
+
+    def build(self):
+        '''
+        Callable method for Block construction.
+        '''
+        super(ReactionParameterBlock, self).build()
+
+        self._reaction_block_class = ReactionBlock
+
+        # Create Phase objects
+        self.Vap = VaporPhase()
+        self.Sol = SolidPhase()
+
+        # Create Component objects
+        self.N2 = Component()
+        self.O2 = Component()
+        self.CO2 = Component()
+        self.H2O = Component()
+        self.Fe2O3 = Component()
+        self.Fe3O4 = Component()
+        self.Al2O3 = Component()
+
+        # Component list subsets
+        self.gas_component_list = Set(initialize=['O2', 'N2', 'CO2', 'H2O'])
```

Comment:
Continuing from above, referring to the component lists in the thermo property packages would have got you these directly.